src/lib/ng-package/entry-point/compile-ngc.transform.ts

```diff
@@ -24,31 +33,40 @@ export const compileNgcTransformFactory = (StylesheetProcessor: typeof Styleshee
       // Compile TypeScript sources
       const { esm2015, declarations } = entryPoint.data.destinationFiles;
       const { basePath, cssUrl, styleIncludePaths } = entryPoint.data.entryPoint;
-      const { moduleResolutionCache, ngccProcessingCache, stylesheetProcessor = new StylesheetProcessor(basePath, cssUrl, styleIncludePaths) } = entryPoint.cache;
-      entryPoint.cache.stylesheetProcessor = stylesheetProcessor;
+      const { moduleResolutionCache, ngccProcessingCache } = entryPoint.cache;
 
-      const ngccProcessor = tsConfig.options.enableIvy
-        ? new NgccProcessor(ngccProcessingCache, tsConfig.project, tsConfig.options, entryPoints)
-        : undefined;
+      if (tsConfig.options.enableIvy !== false && !isDisabled(process.env['NG_BUILD_LIB_LEGACY'])) {
```

Comment:
In my mind the environment variable is backwards. `NG_BUILD_LIB_LEGACY=1` implies you want the legacy pipeline, which lets `isDisabled` return `false`, becoming `true` after the negation and hence this code path is enabled when the `LEGACY` flag is set.

tl;dr; I think this should be
```suggestion
      if (tsConfig.options.enableIvy !== false && isDisabled(process.env['NG_BUILD_LIB_LEGACY'])) {
```

Reply:
You are toght